src/sentry/stacktraces/functions.py

```diff
@@ -0,0 +1,175 @@
+    return [''.join(x) for x in rv]
+
+
+def trim_function_name(function, platform):
+    """Given a function value from the frame's function attribute this returns
+    a trimmed version that can be stored in `function_name`.  This is only used
+    if the client did not supply a value itself already.
+    """
+    if get_behavior_family_for_platform(platform) != 'native':
+        return function
+    if function in ('<redacted>', '<unknown>'):
+        return function
+
+    original_function = function
+    function = function.strip()
+
+    # Ensure we don't operate on objc functions
+    if function.startswith(('[', '+[', '-[')):
+        return function
+
+    # Chop off C++ trailers
+    while 1:
+        match = _cpp_trailer_re.search(function)
+        if match is None:
+            break
+        function = function[:match.start()].rstrip()
+
+    # Because operator<< really screws with our balancing, so let's work
+    # around that by replacing it with a character we do not observe in
+    # `split_func_tokens` or `replace_enclosed_string`.
+    function = function \
+        .replace('operator<<', u'operator⟨⟨') \
+        .replace('operator<', u'operator⟨') \
+        .replace('operator()', u'operator◯')\
+        .replace(' -> ', u' ⟿ ')
+
+    # Remove the arguments if there is one.
+    def process_args(value, start):
+        value = value.strip()
+        if value in ('anonymous namespace', 'operator'):
+            return '(%s)' % value
+        return ''
+    function = replace_enclosed_string(function, '(', ')', process_args)
+
+    # Resolve generic types, but special case rust which uses things like
+    # <Foo as Bar>::baz to denote traits.
+    def process_generics(value, start):
+        # Rust special case
+        if start == 0:
+            return '<%s>' % replace_enclosed_string(value, '<', '>', process_generics)
+        return '<T>'
+    function = replace_enclosed_string(function, '<', '>', process_generics)
+
+    tokens = split_func_tokens(function)
+
+    # find the token which is the function name.  Since we chopped of C++
+    # trailers there are only two cases we care about: the token left to
+    # the -> return marker which is for instance used in Swift and if that
+    # is not found, the last token in the last.
+    #
+    # ["unsigned", "int", "whatever"] -> whatever
+    # ["@objc", "whatever", "->", "int"] -> whatever
+    try:
+        func_token = tokens[tokens.index(u'⟿') - 1]
+    except ValueError:
+        if tokens:
+            func_token = tokens[-1]
+        else:
+            func_token = None
+
+    if func_token:
+        function = func_token.replace(u'⟨', '<') \
+            .replace(u'◯', '()') \
+            .replace(u' ⟿ ', ' -> ')
+
+    # This really should never happen
+    else:
+        function = original_function
+
+    # trim off rust markers
+    function = _rust_hash.sub('', function)
+
+    # trim off windows decl markers
+    return _windecl_hash.sub('\\1', function)
```

Comment:
I know this was in the old code already, but what's the point of replacing with `\1` here? Also, this regex doesn't match the space that usually comes before the `const` qualifier.

Reply:
`\1` is the first capture group which is the function name. The space is effectively implied for the trailers because you need a word boundary and whitespace trimming is done after the fact.